web/main.go

```diff
@@ -161,15 +180,34 @@ func logHandler(h http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+// traceHandler decorates HandlerFunc with context that contains the trace span
+// and automatically closes the span after the request is handled.
+func (s *server) traceHandler(h func(context.Context, http.ResponseWriter, *http.Request)) http.HandlerFunc {
```

Comment:
Is there a reason you don't use https://godoc.org/cloud.google.com/go/trace/traceutil#HTTPHandler?